src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingContext.cs

```diff
@@ -31,40 +45,87 @@ public ResponseCachingContext(HttpContext httpContext, IResponseCache cache)
             Cache = cache;
         }
 
+        internal bool CacheResponse
+        {
+            get
+            {
+                if (_cacheResponse == null)
+                {
+                    // TODO: apparent age vs corrected age value
+                    var responseAge = _responseTime - ResponseHeaders.Date ?? TimeSpan.Zero;
+
+                    _cacheResponse = ResponseStarted && ResponseIsCacheable() && EntryIsFresh(ResponseHeaders, responseAge, verifyAgainstRequest: false);
```

Comment:
Ah, I forgot to update this. It used to be evaluated in several places but now it's only evaluated after OnResponseStarting so ResponseStarted will always be set to true. No need to check for it anymore